x-pack/plugins/monitoring/public/application/pages/no_data/no_data_page.tsx

```diff
@@ -0,0 +1,238 @@
+
+interface NoDataPageSetupDeps {
+  http: any;
+  data: any;
+}
+
+interface SettingsChecker {
+  message: string;
+  api: string;
+  next?: SettingsChecker;
+}
+
+const clusterCheckers: SettingsChecker[] = [
+  {
+    message: 'Checking cluster settings API on production cluster',
+    api: '../api/monitoring/v1/elasticsearch_settings/check/cluster',
+  },
+  {
+    message: 'Checking nodes settings API on production cluster',
+    api: '../api/monitoring/v1/elasticsearch_settings/check/nodes',
+  },
+];
+
+export const NoDataPage = () => {
+  const title = i18n.translate('xpack.monitoring.noData.routeTitle', {
+    defaultMessage: 'Setup Monitoring',
+  });
+
+  const { services } = useKibana<NoDataPageSetupDeps>();
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+
+  const [model, setModel] = useState({
+    errors: [], // errors can happen from trying to check or set ES settings
+    checkMessage: null, // message to show while waiting for api response
+    isLoading: true, // flag for in-progress state of checking for no data reason
+    isCollectionEnabledUpdating: false, // flags to indicate whether to show a spinner while waiting for ajax
+    isCollectionEnabledUpdated: false,
+    isCollectionIntervalUpdating: false,
+    isCollectionIntervalUpdated: false,
+  } as any);
+
+  const { update: updateBreadcrumbs } = useContext(BreadcrumbContainer.Context);
+  updateBreadcrumbs([
+    {
+      'data-test-subj': 'breadcrumbClusters',
+      text: 'Clusters',
+      href: '#/home',
+      ignoreGlobalState: true,
+    },
+  ]);
+
+  const globalState = useContext(GlobalStateContext);
+  initSetupModeState(globalState, services.http);
+
+  // From x-pack/plugins/monitoring/public/views/no_data/model_updater.js
+  const updateModel = useCallback(
+    (properties: any) => {
+      setModel((previousModel: any) => {
+        const updated = { ...previousModel };
+        const keys = Object.keys(properties);
+
+        keys.forEach((key) => {
+          if (Array.isArray(updated[key])) {
+            updated[key].push(properties[key]);
+          } else {
+            updated[key] = properties[key];
+          }
+        });
+
+        return updated;
+      });
+    },
+    [setModel]
+  );
+
+  const getPageData = useCallback(async () => {
+    let catchReason;
+    try {
+      const clusters = await getClusters(services);
+
+      if (clusters && clusters.length) {
+        setShouldRedirect(true);
+        return;
+      }
+    } catch (err) {
+      if (err && err.status === 503) {
+        catchReason = {
+          property: 'custom',
+          message: err.data.message,
+        };
+      }
+    }
+
+    if (catchReason) {
+      updateModel({ reason: catchReason });
+    } else {
+      // TODO not sure if we can add this conditional, but might be required for smooth internal collection enablement
+      // if (!this.isCollectionEnabledUpdating && !this.isCollectionIntervalUpdating) {
```

Comment:
Were you able to figure out if this was necessary?

Reply:
It's not really clear yet. I sometimes get a refresh of "we couldn't enable monitoring" that _might_ be rooted in these conditionals missing. But I haven't been able to cleanly add them without leading to weird dependency cycles.